managed_db.go

```diff
@@ -87,3 +87,18 @@ func (db *DB) SetDiscardTs(ts uint64) {
 	}
 	db.orc.setDiscardTs(ts)
 }
+
+// PreCommitAt will perform prevalidation for CommitAt and return true if commit the transaction can be committed.
+func (txn *Txn) PreCommitAt(commitTs uint64) bool {
+	if !txn.db.opt.managedTxns {
+		panic("Cannot use CanCommitAt with managedDB=false. Use CanCommit instead.")
+	}
+
+	txn.commitTs = commitTs
+
+	txn.db.orc.writeChLock.Lock()
+	defer txn.db.orc.writeChLock.Unlock()
+	txn.precommitTs, txn.precommitConflict = txn.db.orc.newCommitTs(txn)
```

Comment:
calling `newCommitTs` here is pointless because we have already set `precommitTs` and it will mostly be non zero